### PR TITLE
Allow default behavior for ARM64 machines and skip high-memory VMs

### PR DESCRIPTION
**Pull request by 0xba1a**
Use `df` to get the available disk size instead of relying on capability.

Allow Mariner (both 2.0 and 3.0) to use the default kdump config options. Set force_rebuild only on high-mem machines where the crashdump path has to be updated.

Skip testing on high-mem machines for arm64 architecture.

### CODE REVIEW

File: lisa/tools/kdump.py [hunk 1]
```diff
@@ -669,6 +649,42 @@ def config_resource_disk_dump_path(self, dump_path: str) -> None:
         If the system memory size is bigger than 1T, the default size of /var/crash
         may not be enough to store the dump file, need to change the dump path
         """
+
+        """
+        path option is not supported by default initrd. Regenerated initrd will not
+        boot properly in ARM64. So, skip the test if it is ARM64 and Mariner-2.0
+        """
+        if (
+            self.node.os.information.version.major == 2
```
Comment (squirrelsc) [inline lisa/tools/kdump.py:656]: Please make sure the code logic is for Mariner only by check OS type.

**Comment by 0xba1a** [comment]
Testing

- 2.0 ARM64 Image: MicrosoftCBLMariner:cbl-mariner:cbl-mariner-2-arm64:2.20241029.01
    
    ```jsx
    2024-11-06 17:54:53.535[124836087387904][INFO] lisa.RootRunner         KdumpCrash.verify_kdumpcrash_on_random_cpu: PASSED   
    2024-11-06 17:54:53.536[124836087387904][INFO] lisa.RootRunner                   KdumpCrash.verify_kdumpcrash_smp: PASSED   
    2024-11-06 17:54:53.536[124836087387904][INFO] lisa.RootRunner           KdumpCrash.verify_kdumpcrash_single_core: SKIPPED  deployment skipped: ["no available quota found on 'westus3'."], runbook: Environment(name='generated_2', topology='subnet', nodes_raw=None, nodes_requirement=[type:requirement,name:,default:False,count:1,core:1,mem:[2048,],disk:has_resource_disk: None,os_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.Ephemeral,DiskType.PremiumSSDLRS,DiskType.StandardSSDLRS],os_disk_size: [0,],data_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.UltraSSDLRS,DiskType.PremiumV2SSDLRS,DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS],count: [0,],caching: None,iops: [0,],throughput: [0,],size: [0,],max_data_disk_count: [0,],disk_controller_type: allowed:False,items:[DiskControllerType.SCSI],network interface: data_path:allowed:False,items:[NetworkDataPath.Sriov,NetworkDataPath.Synthetic], nic_count:[1,], max_nic_count:[1,], gpu:[0,],f:allowed:True,items:[arch:Arm64,SecurityProfileSettings(extended_schemas={}, type='Security_Profile', security_profile=allowed:False,items:[SecurityProfileType.Standard], encrypt_disk=False, disk_encryption_set_id=''),gen:2],ef:None,NodeSpace(type='requirement')], _original_nodes_requirement=[type:requirement,name:,default:False,count:1,core:1,mem:[2048,],disk:has_resource_disk: None,os_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.Ephemeral,DiskType.PremiumSSDLRS,DiskType.StandardSSDLRS],os_disk_size: [0,],data_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.UltraSSDLRS,DiskType.PremiumV2SSDLRS,DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS],count: [0,],caching: None,iops: [0,],throughput: [0,],size: [0,],max_data_disk_count: [0,],disk_controller_type: allowed:False,items:[DiskControllerType.SCSI],network interface: data_path:allowed:False,items:[NetworkDataPath.Sriov,NetworkDataPath.Synthetic], nic_count:[1,], max_nic_count:[1,], gpu:[0,],f:allowed:True,items:[arch:Arm64,SecurityProfileSettings(extended_schemas={}, type='Security_Profile', security_profile=allowed:False,items:[SecurityProfileType.Standard], encrypt_disk=False, disk_encrypt
    2024-11-06 17:54:53.537[124836087387904][INFO] lisa.RootRunner              KdumpCrash.verify_kdumpcrash_on_cpu32: SKIPPED  skipped: path option is not supported in Mariner-2.0 for ARM64. kdump will not work well on high memory Mariner-2.0 systems
    2024-11-06 17:54:53.537[124836087387904][INFO] lisa.RootRunner test result summary
    2024-11-06 17:54:53.538[124836087387904][INFO] lisa.RootRunner     TOTAL    : 4
    2024-11-06 17:54:53.538[124836087387904][INFO] lisa.RootRunner     QUEUED   : 0
    2024-11-06 17:54:53.539[124836087387904][INFO] lisa.RootRunner     ASSIGNED : 0
    2024-11-06 17:54:53.539[124836087387904][INFO] lisa.RootRunner     RUNNING  : 0
    2024-11-06 17:54:53.540[124836087387904][INFO] lisa.RootRunner     FAILED   : 0
    2024-11-06 17:54:53.540[124836087387904][INFO] lisa.RootRunner     PASSED   : 2
    2024-11-06 17:54:53.540[124836087387904][INFO] lisa.RootRunner     SKIPPED  : 2
    ```
    
- 2.0 AMD64 Image: MicrosoftCBLMariner:cbl-mariner:cbl-mariner-2-gen2:2.20240829.01
    
    ```jsx
    2024-11-06 18:06:40.979[140038202996480][INFO] lisa.RootRunner ________________________________________
    2024-11-06 18:06:40.979[140038202996480][INFO] lisa.RootRunner         KdumpCrash.verify_kdumpcrash_on_random_cpu: PASSED   
    2024-11-06 18:06:40.979[140038202996480][INFO] lisa.RootRunner                   KdumpCrash.verify_kdumpcrash_smp: PASSED   
    2024-11-06 18:06:40.979[140038202996480][INFO] lisa.RootRunner           KdumpCrash.verify_kdumpcrash_single_core: SKIPPED  deployment skipped: ["no available quota found on 'westus3'."], runbook: Environment(name='generated_2', topology='subnet', nodes_raw=None, nodes_requirement=[type:requirement,name:,default:False,count:1,core:1,mem:[2048,],disk:has_resource_disk: None,os_disk_type: allowed:True,items:[DiskType.PremiumSSDLRS,DiskType.StandardHDDLRS,DiskType.Ephemeral,DiskType.StandardSSDLRS],os_disk_size: [0,],data_disk_type: allowed:True,items:[DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS,DiskType.StandardHDDLRS,DiskType.PremiumV2SSDLRS,DiskType.UltraSSDLRS],count: [0,],caching: None,iops: [0,],throughput: [0,],size: [0,],max_data_disk_count: [0,],disk_controller_type: allowed:False,items:[DiskControllerType.NVME,DiskControllerType.SCSI],network interface: data_path:allowed:False,items:[NetworkDataPath.Synthetic,NetworkDataPath.Sriov], nic_count:[1,], max_nic_count:[1,], gpu:[0,],f:allowed:True,items:[arch:x64,gen:2,SecurityProfileSettings(extended_schemas={}, type='Security_Profile', security_profile=allowed:False,items:[SecurityProfileType.SecureBoot,SecurityProfileType.Standard], encrypt_disk=False, disk_encryption_set_id='')],ef:None,NodeSpace(type='requirement')], _original_nodes_requirement=[type:requirement,name:,default:False,count:1,core:1,mem:[2048,],disk:has_resource_disk: None,os_disk_type: allowed:True,items:[DiskType.PremiumSSDLRS,DiskType.StandardHDDLRS,DiskType.Ephemeral,DiskType.StandardSSDLRS],os_disk_size: [0,],data_disk_type: allowed:True,items:[DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS,DiskType.StandardHDDLRS,DiskType.PremiumV2SSDLRS,DiskType.UltraSSDLRS],count: [0,],caching: None,iops: [0,],throughput: [0,],size: [0,],max_data_disk_count: [0,],disk_controller_type: allowed:False,items:[DiskControllerType.NVME,DiskControllerType.SCSI],network interface: data_path:allowed:False,items:[NetworkDataPath.Synthetic,NetworkDataPath.Sriov], nic_count:[1,], max_nic_count:[1,], gpu:[0,],f:allowed:True,items:[arch:x64,gen:2,SecurityProfileSettings(extended_schemas={}, type='Security_Profile', security_profile=all
    2024-11-06 18:06:40.979[140038202996480][INFO] lisa.RootRunner              KdumpCrash.verify_kdumpcrash_on_cpu32: PASSED   
    2024-11-06 18:06:40.980[140038202996480][INFO] lisa.RootRunner test result summary
    2024-11-06 18:06:40.980[140038202996480][INFO] lisa.RootRunner     TOTAL    : 4
    2024-11-06 18:06:40.980[140038202996480][INFO] lisa.RootRunner     QUEUED   : 0
    2024-11-06 18:06:40.980[140038202996480][INFO] lisa.RootRunner     ASSIGNED : 0
    2024-11-06 18:06:40.980[140038202996480][INFO] lisa.RootRunner     RUNNING  : 0
    2024-11-06 18:06:40.980[140038202996480][INFO] lisa.RootRunner     FAILED   : 0
    2024-11-06 18:06:40.980[140038202996480][INFO] lisa.RootRunner     PASSED   : 3
    2024-11-06 18:06:40.980[140038202996480][INFO] lisa.RootRunner     SKIPPED  : 1
    ```
    
- 3.0 ARM64 - MicrosoftCBLMariner:azure-linux-3:azure-linux-3-arm64:3.20241005.01
    
    ```jsx
    2024-11-06 19:06:28.423[133032845764352][INFO] lisa.RootRunner ________________________________________
    2024-11-06 19:06:28.423[133032845764352][INFO] lisa.RootRunner         KdumpCrash.verify_kdumpcrash_on_random_cpu: PASSED   
    2024-11-06 19:06:28.424[133032845764352][INFO] lisa.RootRunner                   KdumpCrash.verify_kdumpcrash_smp: PASSED   
    2024-11-06 19:06:28.424[133032845764352][INFO] lisa.RootRunner           KdumpCrash.verify_kdumpcrash_single_core: SKIPPED  deployment skipped: ["no available quota found on 'westus3'."], runbook: Environment(name='generated_2', topology='subnet', nodes_raw=None, nodes_requirement=[type:requirement,name:,default:False,count:1,core:1,mem:[2048,],disk:has_resource_disk: None,os_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.StandardSSDLRS,DiskType.Ephemeral,DiskType.PremiumSSDLRS],os_disk_size: [0,],data_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.PremiumV2SSDLRS,DiskType.StandardSSDLRS,DiskType.UltraSSDLRS,DiskType.PremiumSSDLRS],count: [0,],caching: None,iops: [0,],throughput: [0,],size: [0,],max_data_disk_count: [0,],disk_controller_type: allowed:False,items:[DiskControllerType.NVME,DiskControllerType.SCSI],network interface: data_path:allowed:False,items:[NetworkDataPath.Sriov,NetworkDataPath.Synthetic], nic_count:[1,], max_nic_count:[1,], gpu:[0,],f:allowed:True,items:[gen:2,arch:Arm64,SecurityProfileSettings(extended_schemas={}, type='Security_Profile', security_profile=allowed:False,items:[SecurityProfileType.Standard], encrypt_disk=False, disk_encryption_set_id='')],ef:None,NodeSpace(type='requirement')], _original_nodes_requirement=[type:requirement,name:,default:False,count:1,core:1,mem:[2048,],disk:has_resource_disk: None,os_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.StandardSSDLRS,DiskType.Ephemeral,DiskType.PremiumSSDLRS],os_disk_size: [0,],data_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.PremiumV2SSDLRS,DiskType.StandardSSDLRS,DiskType.UltraSSDLRS,DiskType.PremiumSSDLRS],count: [0,],caching: None,iops: [0,],throughput: [0,],size: [0,],max_data_disk_count: [0,],disk_controller_type: allowed:False,items:[DiskControllerType.NVME,DiskControllerType.SCSI],network interface: data_path:allowed:False,items:[NetworkDataPath.Sriov,NetworkDataPath.Synthetic], nic_count:[1,], max_nic_count:[1,], gpu:[0,],f:allowed:True,items:[gen:2,arch:Arm64,SecurityProfileSettings(extended_schemas={}, type='Security_Profile', security_profile=allowed:False,items:[SecurityP
    2024-11-06 19:06:28.425[133032845764352][INFO] lisa.RootRunner              KdumpCrash.verify_kdumpcrash_on_cpu32: PASSED   
    2024-11-06 19:06:28.425[133032845764352][INFO] lisa.RootRunner test result summary
    2024-11-06 19:06:28.426[133032845764352][INFO] lisa.RootRunner     TOTAL    : 4
    2024-11-06 19:06:28.426[133032845764352][INFO] lisa.RootRunner     QUEUED   : 0
    2024-11-06 19:06:28.426[133032845764352][INFO] lisa.RootRunner     ASSIGNED : 0
    2024-11-06 19:06:28.426[133032845764352][INFO] lisa.RootRunner     RUNNING  : 0
    2024-11-06 19:06:28.427[133032845764352][INFO] lisa.RootRunner     FAILED   : 0
    2024-11-06 19:06:28.427[133032845764352][INFO] lisa.RootRunner     PASSED   : 3
    2024-11-06 19:06:28.427[133032845764352][INFO] lisa.RootRunner     SKIPPED  : 1
    ```
    
- 3.0 AMD64 - MicrosoftCBLMariner:azure-linux-3:azure-linux-3-gen2:3.20241005.01
    
    ```jsx
    2024-11-06 19:18:04.780[133839578126080][INFO] lisa.RootRunner ________________________________________
    2024-11-06 19:18:04.781[133839578126080][INFO] lisa.RootRunner         KdumpCrash.verify_kdumpcrash_on_random_cpu: PASSED   
    2024-11-06 19:18:04.781[133839578126080][INFO] lisa.RootRunner                   KdumpCrash.verify_kdumpcrash_smp: PASSED   
    2024-11-06 19:18:04.781[133839578126080][INFO] lisa.RootRunner           KdumpCrash.verify_kdumpcrash_single_core: SKIPPED  deployment skipped: ["no available quota found on 'westus3'."], runbook: Environment(name='generated_2', topology='subnet', nodes_raw=None, nodes_requirement=[type:requirement,name:,default:False,count:1,core:1,mem:[2048,],disk:has_resource_disk: None,os_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.Ephemeral,DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS],os_disk_size: [0,],data_disk_type: allowed:True,items:[DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS,DiskType.StandardHDDLRS,DiskType.UltraSSDLRS,DiskType.PremiumV2SSDLRS],count: [0,],caching: None,iops: [0,],throughput: [0,],size: [0,],max_data_disk_count: [0,],disk_controller_type: allowed:False,items:[DiskControllerType.SCSI,DiskControllerType.NVME],network interface: data_path:allowed:False,items:[NetworkDataPath.Synthetic,NetworkDataPath.Sriov], nic_count:[1,], max_nic_count:[1,], gpu:[0,],f:allowed:True,items:[SecurityProfileSettings(extended_schemas={}, type='Security_Profile', security_profile=allowed:False,items:[SecurityProfileType.SecureBoot,SecurityProfileType.Standard], encrypt_disk=False, disk_encryption_set_id=''),arch:x64,gen:2],ef:None,NodeSpace(type='requirement')], _original_nodes_requirement=[type:requirement,name:,default:False,count:1,core:1,mem:[2048,],disk:has_resource_disk: None,os_disk_type: allowed:True,items:[DiskType.StandardHDDLRS,DiskType.Ephemeral,DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS],os_disk_size: [0,],data_disk_type: allowed:True,items:[DiskType.StandardSSDLRS,DiskType.PremiumSSDLRS,DiskType.StandardHDDLRS,DiskType.UltraSSDLRS,DiskType.PremiumV2SSDLRS],count: [0,],caching: None,iops: [0,],throughput: [0,],size: [0,],max_data_disk_count: [0,],disk_controller_type: allowed:False,items:[DiskControllerType.SCSI,DiskControllerType.NVME],network interface: data_path:allowed:False,items:[NetworkDataPath.Synthetic,NetworkDataPath.Sriov], nic_count:[1,], max_nic_count:[1,], gpu:[0,],f:allowed:True,items:[SecurityProfileSettings(extended_schemas={}, type='Security_Profile', security_profile=allowed:False,item
    2024-11-06 19:18:04.781[133839578126080][INFO] lisa.RootRunner              KdumpCrash.verify_kdumpcrash_on_cpu32: PASSED   
    2024-11-06 19:18:04.781[133839578126080][INFO] lisa.RootRunner test result summary
    2024-11-06 19:18:04.781[133839578126080][INFO] lisa.RootRunner     TOTAL    : 4
    2024-11-06 19:18:04.781[133839578126080][INFO] lisa.RootRunner     QUEUED   : 0
    2024-11-06 19:18:04.781[133839578126080][INFO] lisa.RootRunner     ASSIGNED : 0
    2024-11-06 19:18:04.782[133839578126080][INFO] lisa.RootRunner     RUNNING  : 0
    2024-11-06 19:18:04.782[133839578126080][INFO] lisa.RootRunner     FAILED   : 0
    2024-11-06 19:18:04.782[133839578126080][INFO] lisa.RootRunner     PASSED   : 3
    2024-11-06 19:18:04.782[133839578126080][INFO] lisa.RootRunner     SKIPPED  : 1
    ```

File: lisa/tools/kdump.py (hunk 1 above)
Comment (0xba1a) [reply lisa/tools/kdump.py:656]: As this is the `CBLMariner` override-function from the base class, is it really necessary to check the OS type here?

Comment (squirrelsc) [reply lisa/tools/kdump.py:656]: Thank you for pointing out. You are right, it doesn't need.

Comment (squirrelsc) [reply lisa/tools/kdump.py:656]: One more question, should it be like `>= 2`? How about Mariner 3?

Comment (0xba1a) [reply lisa/tools/kdump.py:656]: It's only applicable for 2.0. Mariner-3 is good.
That library loading issue got fixed.

**Comment by squirrelsc** [comment]
@LiliDeng LGTM

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**